pkg/storage/unified/resource/index_mapping.go

```diff
@@ -0,0 +1,176 @@
+package resource
+
+import (
+	"github.com/blevesearch/bleve/v2"
+	"github.com/blevesearch/bleve/v2/mapping"
+	"github.com/grafana/grafana/pkg/apimachinery/utils"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+type IndexedResource struct {
+	Group     string
+	Namespace string
+	Kind      string
+	Name      string
+	Title     string
+	CreatedAt string
+	CreatedBy string
+	UpdatedAt string
+	UpdatedBy string
+	FolderId  string
+	Spec      any
+}
+
+// NewIndexedResource creates a new IndexedResource from a raw resource.
+// rawResource is the raw json for the resource from unified storage.
+func NewIndexedResource(rawResource []byte) (*IndexedResource, error) {
```

Comment:
IMO this might be a good point at which to start adding unit tests

Reply:
If were going to add tests, I'd prefer a few high value ones. Its really easy to make a change that causes no search results show up. So something that tests the whole flow of indexing, run a search query, and validate results.

This PR is big enough, but I can follow up with some tests afterwards.

Reply:
Issue added https://github.com/grafana/grafana-org/issues/230